src/python/pants/backend/rust/target_types.py

```diff
@@ -0,0 +1,30 @@
+# Copyright 2023 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+from pants.engine.target import COMMON_TARGET_FIELDS, MultipleSourcesField, Target, \
+    SingleSourceField
+
+
+class CargoTomlSourceField(SingleSourceField):
+    default = "Cargo.toml"
+
+
+class RustWorkspaceTarget(Target):
+    alias = "rust_workspace"
+    core_fields = (
+        *COMMON_TARGET_FIELDS,
+        CargoTomlSourceField,
+    )
+
+
+class RustPackageSourcesField(MultipleSourcesField):
+    default = ("Cargo.toml", "src/**/*.rs", "tests/**/*.rs")
+    uses_source_roots = False
+
+
+class RustPackageTarget(Target):
```

Comment:
This should have a `help` attribute as well. For example: https://github.com/pantsbuild/pants/blob/584679af6432835a5bfd2b9784a1baeacf5792a9/src/python/pants/backend/go/target_types.py#L407-L414